Sourcery/Parsing/FileParser.swift

```diff
@@ -307,6 +312,73 @@ extension FileParser {
         return true
     }
 
+    fileprivate func extractGenericsDeclaration(source: String) -> String {
+        var declaration = source
+        if let whereStart = declaration.index(of: "where") {
+            declaration = String(declaration[declaration.startIndex ..< whereStart])
+        }
+
+        if let start = source.index(of: "<") {
```

Comment:
this seems to work on `class MyClass: SuperClass<Generic>` when it shouldn't, so we should check for inheritance part and drop it first.